tests/unit/custom-actions-test.js

```diff
@@ -88,4 +88,33 @@ module('customAction()', function (hooks) {
     let response = await apiAction(user, { method: 'POST', path: 'like' });
     assert.deepEqual(response, { custom: 'buildURL' });
   });
+
+  test('snapshot can receive adapterOptions', async function (assert) {
+    class UserAdapter extends RESTAdapter {
+      buildURL(modelName, id, snapshot) {
+        if (snapshot?.adapterOptions) {
```

Comment:
```suggestion
        if (snapshot.adapterOptions.test === true) {
```

let's be specific here though and it's fine if this throws an error if `snapshot` or `adapterOptions` doesn't exist

Reply:
It would fail using the prepare function though since adapterOptions` does not exist when we [try and find the record](https://github.com/SkoebaSteve/ember-api-actions/blob/custom-adapter-options/tests/unit/custom-actions-test.js#L24).

> let's be specific here though and it's fine if this throws an error if snapshot or adapterOptions doesn't exist

Since it's just for the test, I felt the conditional chain was ok here. Are you saying it's better to allow it to fail in the test?

Alternatively I could also not use the prepare function for this scenario.

Reply:
my reason for this is that while `RESTAdapter` currently does not add any `adapterOptions` itself, nothing is preventing it from doing something like that in the future. I'd rather be specific and check that we see exactly what we passed in ourselves :)

Reply:
```suggestion
        if (snapshot?.adapterOptions.test === true) {
```

So would this be ok for you? Just so I can easily reuse the prepare function :)

Reply:
sure

Reply:
Done!